src/applications/vaos/tests/project-cheetah/components/SecondDosePage.unit.spec.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import moment from 'moment';
+import { expect } from 'chai';
+import userEvent from '@testing-library/user-event';
+
+import { createTestStore, renderWithStoreAndRouter } from '../../mocks/setup';
+
+import SecondDosePage from '../../../project-cheetah/components/SecondDosePage';
+
+const initialState = {
+  featureToggles: {
+    vaOnlineSchedulingCancel: true,
+    vaOnlineSchedulingProjectCheetah: true,
+  },
+};
+
+describe('VAOS vaccine flow <SecondDosePage>', () => {
+  const start = moment();
+  const store = createTestStore({
+    ...initialState,
+    projectCheetah: {
+      newBooking: {
+        data: {
+          vaFacility: 'var983',
+          clinicId: '455',
+          date1: [start.format()],
+        },
+        availableSlots: [
+          {
+            start: start.format(),
+            end: start
+              .clone()
+              .add(30, 'minutes')
+              .format(),
+          },
+        ],
+        clinics: {},
+        facilities: [
+          {
+            id: 'var983',
+            name: 'Cheyenne VA Medical Center',
+          },
+        ],
+      },
+    },
+  });
+
+  it('should show second vaccine dose information', async () => {
+    const screen = renderWithStoreAndRouter(<SecondDosePage />, {
+      store,
+    });
+
+    expect(await screen.getByText(/Plan your second dose/i)).to.have.tagName(
+      'h1',
+    );
+    expect(
+      screen.getByText(
+        /You'll need to return to the Cheyenne VA Medical Center/i,
+      ),
+    ).to.be.ok;
+    expect(screen.getByText('Moderna')).to.have.tagName('h2');
+    expect(screen.getByText('Pfizer')).to.have.tagName('h2');
+    expect(screen.baseElement).to.contain.text(
+      `If you receive your first dose on ${start.format(
+        'dddd, MMMM DD, YYYY',
+      )}`,
+    );
+    expect(
+      screen.getByText(
+        new RegExp(
+          `Return for your second dose after ${start
+            .add(21, 'days')
+            .format('dddd, MMMM DD, YYYY')}`,
+          'i',
+        ),
+      ),
+    ).to.be.ok;
+    expect(
+      screen.getByText(
+        new RegExp(
+          `Return for your second dose after ${moment()
+            .add(28, 'days')
+            .format('dddd, MMMM DD, YYYY')}`,
+          'i',
+        ),
+      ),
+    ).to.be.ok;
```

Comment:
It would also be nice to have a check here that if you click Continue, you go to the correct next url. I think other tests for pages in the cheetah flow have something like that you can start from.

Reply:
@jbalboni I referenced the other unit tests that checks for the next cheetah flow URL and applied the expect to this test. But what is expected fails.  Would you take a look to see what i did wrong?

Reply:
That was a weird one! The problem is the state you created for the store is missing some default data, and this is causing an error in the reducer where the next page is calculated. For some reason this error isn't output anywhere, I found it by putting a bunch of console.log statements at different steps and walking through the action and reducer code.

I fixed it by adding `previousPages: {},` to the `newBooking` object you're using to create the store (https://github.com/department-of-veterans-affairs/vets-website/pull/16033/files#diff-eb060665ccf32c6c8b44ce06fa0409332af6ad8f75b652fc53ed6c7bad0caa6eR22).

Reply:
I would not have guess the store needs `previousPage` object to make the next flow URL to work.